dialogue-core/src/test/java/com/palantir/dialogue/core/RetryingChannelTest.java

```diff
@@ -169,7 +169,50 @@ public void retries_503s() throws Exception {
     }
 
     @Test
-    public void retries_429s_when_requested() throws Exception {
+    public void retries_308s() throws Exception {
+        Response mockResponse = mock(Response.class);
+        when(mockResponse.code()).thenReturn(308);
+        when(channel.execute(any(), any())).thenReturn(Futures.immediateFuture(mockResponse));
+
+        Channel retryer = new RetryingChannel(
+                channel,
+                "my-channel",
+                3,
+                Duration.ZERO,
+                ClientConfiguration.ServerQoS.AUTOMATIC_RETRY,
+                ClientConfiguration.RetryOnTimeout.DISABLED);
+        ListenableFuture<Response> response = retryer.execute(TestEndpoint.POST, REQUEST);
+        assertThat(response).isDone();
+        assertThat(response.get())
+                .as("After retries are exhausted the 308 response should be returned")
+                .isSameAs(mockResponse);
+        verify(channel, times(4)).execute(TestEndpoint.POST, REQUEST);
+    }
+
+    @Test
+    public void propagates_308s_when_requested() throws Exception {
+        Response mockResponse = mock(Response.class);
+        when(mockResponse.code()).thenReturn(308);
+        when(channel.execute(any(), any())).thenReturn(Futures.immediateFuture(mockResponse));
+
+        Channel retryer = new RetryingChannel(
+                channel,
+                "my-channel",
+                3,
+                Duration.ZERO,
+                // Okay. This is odd. It doesn't say 308, but we consider it the same as other server QoS exceptions.
```

Comment:
Thoughts on this?

Reply:
Hmmm I think this was only added for the time-series folks, I don't think they use 308s at all.  To that end, I think I'd make it do what it says on the tin, and only propagate 429 and 503.

Reply:
Ya I think I agree with dan here

Reply:
That's fair, the code will be a bit crufty, but it will be less astonishing for consumers.

Reply:
Updated, I tried to make it as readable as possible.